adapters/dynamoDB_adapters/src/main/java/org/atomhopper/dynamodb/query/SQLToNoSqlConverter.java

```diff
@@ -0,0 +1,311 @@
+package org.atomhopper.dynamodb.query;
+
+import com.unboundid.ldap.sdk.Filter;
+import com.unboundid.ldap.sdk.LDAPException;
+import org.apache.commons.lang.StringUtils;
+
+import java.util.*;
+
+/**
+ * This class creates the search SQL and corresponding SQL parameters for a given set of query parameters.
+ *
+ * By default, this class assumes all categories are treated the same and stored in a single variable-length array.
+ *
+ * This class can also be customized to map specific categories to specific columns in the DB, allowing for higher
+ * performance when searching on these categories.
+ *
+ * To configure this functionality pass in the following:
+ *
+ * <ul>
+ *     <li>Map[String, String] - a map of pairs where the key is the prefix found in the atom category and the
+ *     value is the name of the SQL text column.</li>
+ *     <li>String - the String which separates the prefix from the value within the atom category.  E.g.,
+ *     for the category value of "tid:1234", if ':' is the mark, then "tid" is the prefix.</li>
+ * </ul>
+ *
+ * An example configuration might be:
+ *
+ * A map of { "tid" => "tenantid", "type" => "eventtype" } with a mark of ":".
+ *
+ * This maps the following atom categories:
+ *
+ * <ul>
+ *     <li>"tid:1234" => enter "1234" into the "tenantid" column</li>
+ *     <li>"type:lbaas.usage" => enter "lbaas.usage" into the "eventtype" column</li>
+ * </ul>
+ */
+public class SQLToNoSqlConverter {
```

Comment:
Again, I doubt if all the methods of this class are used?

Reply:
Yes sone methods need to be removed as they may not be used